### PR TITLE
updates for terraform deployment from automation

### PR DESCRIPTION
**Pull request by 0x0I**
* add explicit *helm/k8s* providers for testnet deployments based on designated `k8s_context` and ensure kube config path is set
* fix archive image launch bug (remove recently added entrypoint override)
* fix HelmRelease dirtyWhen triggers

**Test:** Buildkite CI

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: automation/scripts/generate-keys-and-ledger.sh [hunk 1]
```diff
@@ -301,7 +301,7 @@ PROMPT_KEYSETS="${PROMPT_KEYSETS}n
 # Handle passing the above keyset info into interactive 'coda-network genesis' prompts
 while read input
 do echo "$input"
-  sleep 1
+  sleep 2
```
Comment (0x0I) [inline automation/scripts/generate-keys-and-ledger.sh:304]: fixes key-generation error i'm experiencing locally which looks to be due to some sort of race condition

File: automation/terraform/modules/kubernetes/testnet/helm.tf [hunk 2]
```diff
@@ -168,6 +163,8 @@ locals {
 # Cluster-Local Seed Node
 
 resource "kubernetes_role_binding" "helm_release" {
+  # provider   = helm.testnet_deploy
```
Comment (0x0I) [inline automation/terraform/modules/kubernetes/testnet/helm.tf:166]: uncomment

File: automation/terraform/modules/kubernetes/testnet/helm.tf [hunk 3]
```diff
@@ -237,12 +240,14 @@ resource "helm_release" "snark_workers" {
...
+  provider   = helm.testnet_deploy
+
   count       = var.archive_node_count
   
-  name        = "archive-node-${count.index}"
+  name        = "archive-node-${count.index + 1}"
   repository  = local.use_local_charts ? "" : local.mina_helm_repo
   chart       = local.use_local_charts ? "../../../../helm/archive-node" : "archive-node"
-  version     = "0.3.3"
+  version     = "0.4.6"
```
Comment (AryehAlexander) [inline automation/terraform/modules/kubernetes/testnet/helm.tf:250]: The other versions are 0.4.5 just want to make sure this is correct

**Review by AryehAlexander — APPROVED** [review]
:+1: lgtm, just one double check about versions

File: automation/terraform/modules/kubernetes/testnet/helm.tf (hunk 3 above)
Comment (0x0I) [reply automation/terraform/modules/kubernetes/testnet/helm.tf:250]: Yep, all good here - note: the versions across charts won't necessarily stay in sync

**Comment by 0x0I** [comment]
Re-targeted at `compatible` via #7573

**Pull request closed without merging**